packages/@webex/plugin-cc/test/unit/spec/AgentConfigService/AgentConfigService.ts

```diff
@@ -0,0 +1,194 @@
+import { WebexSDK } from '../../../../src/types';
+import HttpRequest from '../../../../src/HttpRequest';
+import AgentConfigService from '../../../../src/AgentConfigService/AgentConfigService';
+
+jest.mock('../../../../src/HttpRequest', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      request: jest.fn(),
+    };
+  });
+});
+
+describe('AgentConfigService', () => {
+  let agentConfigService: AgentConfigService;
+  let mockHttpRequest: jest.Mocked<HttpRequest>;
+  const mockWebexSDK: WebexSDK = {
+    logger: {
+      log: jest.fn(),
+    },
+  } as unknown as WebexSDK;
+  const mockWccAPIURL = 'https://api.example.com/';
+  const mockAgentId = 'agent123';
+  const mockOrgId = 'org123';
+
+  beforeEach(() => {
+    mockHttpRequest = new HttpRequest(mockWebexSDK) as jest.Mocked<HttpRequest>;
+    agentConfigService = new AgentConfigService(mockAgentId, mockOrgId, mockWebexSDK, mockWccAPIURL);
+    agentConfigService.requestInstance = mockHttpRequest;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getUserUsingCI', () => {
+    it('should return AgentResponse on success', async () => {
+      const mockResponse = {
+        statusCode: 200,
+        body: {
+          firstName: 'John',
+          lastName: 'Doe',
+          agentProfileId: 'profile123',
+          email: 'john.doe@example.com',
+          teamIds: ['team1', 'team2'],
+        },
+      };
+      mockHttpRequest.request.mockResolvedValue(mockResponse);
+
+      const result = await agentConfigService.getUserUsingCI();
+
+      expect(result).toEqual(mockResponse.body);
+      expect(mockWebexSDK.logger.log).toHaveBeenCalledWith('getUserUsingCI api success.');
```

Comment:
We should have expect in all these tests for request method being invoked with the arguments passed in AgentConfigService methods.

Reply:
Taken Care!